src/summary.rs

```diff
@@ -45,9 +51,67 @@ impl Summary {
         Self {
             issues: issues.to_vec(),
             data,
+            flake_lock_path,
+            flake_check_config,
         }
     }
 
+    pub fn console_log_errors(&self) -> Result<(), FlakeCheckerError> {
+        let file = self.flake_lock_path.to_string_lossy();
+
+        if self.issues.is_empty() {
+            println!("The Determinate Nix Flake Checker scanned {file} and found no issues");
+        } else {
+            let level = if self.flake_check_config.fail_mode {
+                "error"
+            } else {
+                "warning"
+            };
+
+            for issue in self.issues.iter() {
+                let input = &issue.input;
+
+                let message: Option<String> = match &issue.kind {
+                    IssueKind::Disallowed(disallowed) => {
+                        if self.flake_check_config.check_supported {
+                            let reference = &disallowed.reference;
+                            Some(format!(
+                            "the `{input}` input uses the non-supported Git branch `{reference}` for Nixpkgs"
+                        ))
```

Comment:
Rustfmt chokes on these expressions with long lines, so you'll sadly need to hand-format them :(
```suggestion
                            Some(format!(
                                "the `{input}` input uses the non-supported Git branch `{reference}` for Nixpkgs"
                            ))
```